### PR TITLE
Improve docs of "pygmt.grd2cpt" and "pygmt.makecpt"

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

This PR aims to fix typos in the docstrings for [pygmt.grd2cpt](https://www.pygmt.org/dev/api/generated/pygmt.grd2cpt.html).
Regarding the `color_model` parameter or [**F**](https://docs.generic-mapping-tools.org/dev/grd2cpt.html#f) also a upstream PR seems needed (at least adding a space and escaping it), but first I have to be sure, that my changes are correct 🙂.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Review by seisman — APPROVED** [review]

File: pygmt/src/grd2cpt.py [hunk 1]
```diff
@@ -126,14 +127,14 @@ def grd2cpt(grid, **kwargs):
         also :gmt-docs:`cookbook/features.html#manipulating-cpts`.
     output : str
         Optional parameter to set the file name with extension .cpt to store
-        the generated CPT file. If not given or False [Default], saves the CPT
-        as the session current CPT.
+        the generated CPT file. If not given or ``False`` [Default], saves the
+        CPT as the session current CPT.
```
Comment (michaelgrund) [inline pygmt/src/grd2cpt.py:131]:
```suggestion
        CPT as the current session CPT.
```
In my opinion the words should switched here.

Comment (seisman) [reply pygmt/src/grd2cpt.py:131]: I think both "session current CPT" and "current session CPT" are inaccurate. The key point is that, GMT has several hierarchical levels (https://docs.generic-mapping-tools.org/dev/cookbook/features.html#gmt-modern-mode-hierarchical-levels), thus a "current CPT" could be the current CPT of a session/figure/subplot/panel/inset, depending  in which level the `makecpt`/`grd2cpt` module is called.

Comment (yvonnefroehlich) [reply pygmt/src/grd2cpt.py]: I have made some changes to the docstrings of `grd2cpt` and `makecpt`. @seisman and @michaelgrund do you feel this is an improvement?

File: pygmt/src/makecpt.py [hunk 2]
```diff
@@ -110,8 +115,9 @@ def makecpt(**kwargs):
         also :gmt-docs:`cookbook/features.html#manipulating-cpts`.
     output : str
         Optional. The file name with extension .cpt to store the generated CPT
-        file. If not given or False [Default], saves the CPT as the session
-        current CPT.
+        file. If not given or ``False`` [Default], saves the CPT as the current
+        CPT of the session, figure, subplot, panel, or inset depending on which
+        level :func:`pygmt.grd2cpt` is called.
```
Comment (seisman) [inline pygmt/src/makecpt.py:120]:
```suggestion
        level :func:`pygmt.makecpt` is called.
```

**Review by seisman — APPROVED** [review]
Looks good to me except a typo.

**Pull request merged**